temporal/api/cloud/nexus/v1/message.proto

```diff
@@ -0,0 +1,86 @@
+    // The name of the endpoint. Must be unique within an account.
+    // The name must match `[a-zA-Z_][a-zA-Z0-9_]*`.
+    // This field is mutable.
+    string name = 1;
+
+    // Indicates where the endpoint should forward received nexus requests to. 
+    EndpointTargetSpec endpoint_target_spec = 2;
+
+    // The set of policies (e.g. authorization) for the endpoint. Each request's caller
+    // must match with at least one of the specs to be accepted by the endpoint.
+    // This field is mutable.
+    repeated EndpointPolicySpec policy_specs = 3;
+
+    // Optional description. Can be in markdown for rendering in the UI.
+    // Note that this is an experimental field that may be deprecated in the future
+    string description = 4;
+}
+
+message EndpointTargetSpec {
+    // The type of target that this endpoint routes Nexus requests to.
+    // Possible values are: cloud_task_queue
+    string target_type = 1;
+
+    // A target spec for routing nexus requests to a specific cloud namespace worker.
+    CloudWorkerTargetSpec cloud_worker_target_spec = 2;
+}
+
+message CloudWorkerTargetSpec {
+    // The target cloud namespace to route requests to. Namespace must be in same account as the endpoint. This field is mutable.
+    string namespace = 1;
+
+    // The task queue on the cloud namespace to route requests to. This field is mutable.
+    string task_queue = 2;
+}
+
+message EndpointPolicySpec {
+    // The policy set on this endpoint authorization spec.
+    // Possible values are: allowed_cloud_namespace
+    string policy_type = 1;
+
+    // A policy spec that allows one caller namespace to access the endpoint. 
+    AllowedCloudNamespacePolicySpec allowed_cloud_namespace_policy_spec = 2;
+}
+
+message AllowedCloudNamespacePolicySpec {
+    // The namespace that is allowed to call into this endpoint. Calling namespace must be in same account as the endpoint.
+    string namespace = 1;
+}
+
+// An endpoint that receives and then routes Nexus requests
+message Endpoint {
+    // The id of the endpoint. This is generated by the server and is immutable.
+    string id = 1;
+
+    // The current version of the endpoint specification.
+    // The next update operation must include this version.
+    string resource_version = 2;
+
+    // The endpoint specification.
+    EndpointSpec spec = 3;
+
+    // The current state of the endpoint.
+    // Possible values: activating, activationfailed, active, updating, updatefailed, deleting, deletefailed, deleted
+    // For any failed state, reach out to Temporal Cloud support for remediation.
+    string state = 4;
+
+    // The id of any ongoing async operation that is creating, updating, or deleting the endpoint, if any.
+    string async_operation_id = 5;
+
+    // The date and time when the endpoint was created.
+    google.protobuf.Timestamp created_time = 6;
+
+    // The date and time when the endpoint was last modified.
+    google.protobuf.Timestamp last_modified_time = 7;
```

Comment:
@mastermanu can we also show the identity that created and modified this endpoint?

Reply:
i think we can, although we'd probably want to consider adding that across all the objects we have (e.g. users, namespaces, etc).

would probably consider adding that as a separate effort.